utils/yoke/src/yoke.rs

```diff
@@ -480,50 +471,183 @@ where
     }
 }
 
-impl<Y: for<'a> Yokeable<'a>, T: ?Sized> Clone for Yoke<Y, Option<Rc<T>>>
-where
-    for<'a> <Y as Yokeable<'a>>::Output: Clone,
-{
-    fn clone(&self) -> Self {
+impl<Y: for<'a> Yokeable<'a>, C: CloneableCart> Yoke<Y, C> {
+    /// Allows one to "project" a yoke to perform a transformation on the data, potentially
+    /// looking at a subfield, and producing a new yoke. This will clone the cart, and the provided
+    /// transformation is only allowed to use data known to be borrowed from the cart.
+    ///
+    /// This currently takes an additional `&()` parameter as a workaround to
```

Comment:
If you care deeply about people not touching it you could just use `PhantomData<&'a ()>` instead.

Reply:
It doesn't matter if people read it, but @eddyb had suggested the same thing so sounds like a good idea